### PR TITLE
Build for multiple platforms

### PR DESCRIPTION
**Pull request by AlexKnauth**
This PR adds support for non-Windows platforms, and adds a Github-Actions Workflow to build it for Windows, Mac, and Linux, based on the Github-Actions Workflow used by the Modding API.

Tested on Mac.

Edit: now tested on Mac, Windows, and Linux through WSL.

### CODE REVIEW

**Comment by Yurihaia** [comment]
Thank you! ~~Before I merge this, could you check with the speedrun.com mods for HK to make sure this is ok? This mod is allowed to be used for IL runs so I want to have them review it as well.~~ (did this myself) It looks good to me though.

Just to make sure, can you describe what the release process is? I'm not quite sure looking at the build.yaml file. From what I can tell its just "create a tag" creates a new release but I'm not quite sure I actually want that to be automated.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+name: build
+
+on: [push, pull_request]
+
+jobs:
+  build-win:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```
Comment (Yurihaia) [inline .github/workflows/build.yaml]: Also I believe each of these `@v3`s should be changed to `@v4` to use NodeJS 20 (because NodeJS 16 is deprecated)

**Comment by AlexKnauth** [comment]
My test release here:
https://github.com/AlexKnauth/MiniSavestates/releases/tag/test3

Was created with these commands, after committing:
```sh
git tag -a test3 -m "test 3: actions checkout v4"
git push origin test3
```

Edit:
My test release here:
https://github.com/AlexKnauth/MiniSavestates/releases/tag/test4

Was created with these commands:
```sh
git tag test4
git push origin test4
```

**Comment by AlexKnauth** [comment]
I asked in the Hollow Knight Speedrun Community discord channel `#src-discussion`:
https://discord.com/channels/772964112908156938/822612545792639026/1211795811432665098
and @ManicJamie responded there.

**Comment by Yurihaia** [comment]
There are still NodeJS 16 deprecation warnings in the build so could you resolve those?

**Comment by AlexKnauth** [comment]
Updated versions of all dependencies mentioned in the NodeJS 16 deprecation warnings.

**Comment by Yurihaia** [comment]
https://github.com/Yurihaia/MiniSavestates/actions/runs/8676717657/job/23791494526 the build script is failing for some reason.

**Comment by Yurihaia** [comment]
I ended up changing it to use a private GH repository so that I won't need to rely on a third party website.

**Pull request merged**